homeassistant/components/camera/canary.py

```diff
@@ -0,0 +1,108 @@
+"""
+Support for Canary Security Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.canary/
+"""
+import logging
+
+import requests
+
+from homeassistant.components.camera import Camera
+from homeassistant.components.canary import DATA_CANARY
+
+DEPENDENCIES = ['canary']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MOTION_START_TIME = "motion_start_time"
+ATTR_MOTION_END_TIME = "motion_end_time"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Canary sensors."""
+    data = hass.data[DATA_CANARY]
+    devices = []
+
+    for location in data.locations:
+        entries = data.get_motion_entries(location.location_id)
+        if entries:
+            devices.append(CanaryCamera(data, location.location_id))
+
+    add_devices(devices, True)
+
+
+class CanaryCamera(Camera):
+    """An implementation of a Canary security camera."""
+
+    def __init__(self, data, location_id):
+        """Initialize a Canary security camera."""
+        super().__init__()
+        self._data = data
+        self._location_id = location_id
+
+        self._location = None
+        self._last_entry = None
+        self._image_content = None
+        self._force_update = False
+
+        self.update()
+
+    def camera_image(self):
+        """Return bytes of camera image."""
+        return self._image_content
+
+    @property
+    def name(self):
+        """Return the name of this device."""
+        return self._location.name
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return not self._location.is_private
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        if self._last_entry is None:
+            return None
+
+        return {
+            ATTR_MOTION_START_TIME: self._last_entry.start_time,
+            ATTR_MOTION_END_TIME: self._last_entry.end_time,
+        }
+
+    @property
+    def force_update(self) -> bool:
```

Comment:
Just want to explain a bit why I'm doing this here ...

When you click camera image from UI, on the top left, there a time that indicates how long ago the state was changed. Since the state of the camera is always gonna be the same, the time basically means when HASS was restarted

So instead of that what I want is to show how long ago a motion was detected by Canary camera. Right now, I'm only getting camera image from Canary if there's motion detected by the camera. So whenever I see there's motion detected, I pull the image and cache it in self._iamge_content (see update()). Then I want update the time in top left.

The only way I found that can achieve this is to make force_update return True when I update the image.

Please let me know if there's better way to do this

Reply:
Agree, I removed the change related to force update